includes/orders/functions/orders.php

```diff
@@ -336,8 +341,14 @@ function edd_destroy_order( $order_id = 0 ) {
  */
 function edd_update_order( $order_id = 0, $data = array() ) {
 	$orders = new EDD\Database\Queries\Order();
+	$update = $orders->update_item( $order_id, $data );
+
+	if ( ! empty( $data['customer_id'] ) ) {
+		$customer = new EDD_Customer( $data['customer_id'] );
+		$customer->recalculate_stats();
+	}
```

Comment:
This is probably OK for now, but I have a small concern.

* Putting this here requires that `edd_update_order()` is always the function to call in order to ensure that a customer's stats get recounted when updating an order. If the `update_item()` method is called elsewhere, it won't get the same treatment.
* The customer's stats will _always_ get recounted even if data being changed in the order results in no change to the customer.

Could this maybe be hooked into a `_transition_` type action instead. Like maybe when `$data['status']` changes into or out of a status that would necessitate altering the customer stats.

Again... this good, but _perhaps_ not _completely_ optimized.

Reply:
It's a great question, and I'm honestly not sure. I feel like there are plenty of places that `edd_update_order` is used, but not every usage includes the customer ID. As I was looking for places where the order status might change, and tucking the recalculation call in each, it seemed like a lot. Putting it here felt like the logical place to me because it was really the last gateway, but I think it's quite likely the recalculation could be running more than once in the process of changing an order because of it now being in multiple places. Definitely worth looking into more.